crates/uv-python/src/downloads.rs

```diff
@@ -555,63 +560,73 @@ impl ManagedPythonDownload {
             return Ok(DownloadResult::AlreadyAvailable(path));
         }
 
-        let filename = url.path_segments().unwrap().last().unwrap();
-        let ext = SourceDistExtension::from_path(filename)
-            .map_err(|err| Error::MissingExtension(url.to_string(), err))?;
-        let (reader, size) = read_url(&url, client).await?;
-
-        let progress = reporter
-            .as_ref()
-            .map(|reporter| (reporter, reporter.on_download_start(&self.key, size)));
-
-        // Download and extract into a temporary directory.
-        let temp_dir = tempfile::tempdir_in(scratch_dir).map_err(Error::DownloadDirError)?;
-
-        debug!(
-            "Downloading {url} to temporary location: {}",
-            temp_dir.path().simplified_display()
+        // We improve compatibility by using neither the URL-encoded `%2B` nor the `+` it decodes
+        // to.
```

Comment:
As in, compatibility with file systems? If so, can you add that for clarity? I wasn't sure what we were trying to be compatible with here.

Reply:
Yes, clarified